src/output/workspace-impl.cpp

```diff
@@ -230,10 +230,14 @@ class output_layer_manager_t
     /** Precondition: view is in some sublayer */
     void bring_to_front(wayfire_view view)
     {
-        view->damage();
-
         auto sublayer = get_view_sublayer(view);
         assert(sublayer);
+
+        for (auto view : sublayer->views)
+        {
+            view->damage();
```

Comment:
Now that we're on the topic, we should also damage child views (typically these are dialogs). And we should do that in each function in this file (instead of just view->damage()) ...

Reply:
done.